gallery/gallery-war/src/main/java/org/silverpeas/components/gallery/web/AbstractGalleryResource.java

```diff
@@ -186,24 +204,108 @@ protected Response getMediaContent(final MediaType expectedMediaType, final Stri
+      String mediaMimeType = getHttpRequest().getParameter("forceMimeType");
+      if (StringUtil.isNotDefined(mediaMimeType)) {
+        mediaMimeType = ((InternalMedia) media).getFileMimeType().getMimeType();
+      }
+
+      int length = (int) Files.size(mediaPath);
+      int start = 0;
+      int end = length - 1;
+
+      String range = defaultStringIfNotDefined(getHttpServletRequest().getHeader("Range"), "");
+      Matcher matcher = RANGE_PATTERN.matcher(range);
+      boolean isPartialRequest = matcher.matches();
+
+      if (isPartialRequest) {
+        String startGroup = matcher.group("start");
+        start = startGroup.isEmpty() ? start : Integer.valueOf(startGroup);
+        start = start < 0 ? 0 : start;
+
+        String endGroup = matcher.group("end");
+        end = endGroup.isEmpty() ? end : Integer.valueOf(endGroup);
+        end = end > length - 1 ? length - 1 : end;
+      }
+
+      final int finalStart = start;
+      final int finalContentLength = end - finalStart + 1;
+
+      final Response.ResponseBuilder responseBuilder;
+      if (isPartialRequest) {
+
+        // Handling here a partial response (pseudo streaming)
+
+        final HttpServletResponse response = getHttpServletResponse();
+        response.setBufferSize(BUFFER_LENGTH);
+
+        responseBuilder =
+            Response.status(Status.PARTIAL_CONTENT).entity((StreamingOutput) output -> {
+
+              int bytesRead;
+              int bytesLeft = finalContentLength;
+              ByteBuffer buffer = ByteBuffer.allocate(BUFFER_LENGTH);
+
+              try (SeekableByteChannel input = Files.newByteChannel(mediaPath, READ)) {
+                input.position(finalStart);
+                while ((bytesRead = input.read(buffer)) != -1 && bytesLeft > 0) {
+                  buffer.clear();
+                  output.write(buffer.array(), 0, bytesLeft < bytesRead ? bytesLeft : bytesRead);
+                  bytesLeft -= bytesRead;
+                }
+              } catch (IOException ioe) {
+                SilverLogger.getLogger(AbstractGalleryResource.class).warn(
+                    "client stopping the streaming HTTP Request of media content represented by " +
+                        "''{0}'' identifier (original message ''{1}'')", media.getId(),
+                    ioe.getMessage());
+              }
+            }).header("Accept-Ranges", "bytes")
+              .header("ETag", media.getId())
+              .lastModified(Date.from(Files.getLastModifiedTime(mediaPath).toInstant()))
+              .expires(Date.from(Instant.ofEpochMilli(System.currentTimeMillis() + EXPIRE_TIME)))
+              .header("Content-Range", String.format("bytes %s-%s/%s", start, end, length));
+
+      } else {
+
+        // Handling here a full response
+
+        responseBuilder = Response.ok((StreamingOutput) output -> {
+          try (final InputStream mediaStream = FileUtils.openInputStream(file)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1141)